### PR TITLE
Sync client CA to `InternalSecret` and use it in `shoots/adminkubeconfig` subresource

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area dev-productivity usability
/kind enhancement

**What this PR does / why we need it**:

This PR makes use of the new `InternalSecret` API for syncing the client CA to the garden cluster. The `shoots/adminkubeconfig` subresource retrieves it from there instead of using the `ShootState` resource.
It falls back to the `ShootState` if the `InternalSecret` has not been synced yet.

Once, the fallback has been dropped in the next release, there is no requirement for keeping the `ShootState` for all shoots all the time, which is a prerequisite for GEP-22 (https://github.com/gardener/gardener/issues/8073).

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/7999

**Special notes for your reviewer**:

Co-Authored-By: @rfranzke

Based on https://github.com/gardener-community/hackathon/blob/main/2023-05_Leverkusen/README.md#-introducing-internalsecret-resource-in-gardener-api / https://github.com/rfranzke/gardener/tree/hackathon/internal-secrets

**Release note**:

```feature operator
Gardener uses an `InternalSecret` per Shoot for syncing the client CA to the project namespace in the garden cluster (named `<shoot-name>.ca-client`). The `shoots/adminkubeconfig` subresource signs short-lived client certificates by retrieving the CA from the `InternalSecret`.
```

### CODE REVIEW

File: charts/gardener/controlplane/charts/application/templates/validatingwebhook-admission-controller.yaml [hunk 1]
```diff
@@ -138,17 +138,20 @@ webhooks:
     operations:
     - CREATE
     resources:
+    - backupbuckets
```
Comment (rfranzke) [inline charts/gardener/controlplane/charts/application/templates/validatingwebhook-admission-controller.yaml]: Now `BackupBucket`s are no longer called on `DELETE`. Was this change intentional?

File: pkg/utils/gardener/shoot.go [hunk 2]
```diff
@@ -230,6 +230,13 @@ func GetShootProjectSecretSuffixes() []string {
 	}
 }
 
+// GetShootProjectInternalSecretSuffixes returns the list of shoot-related project internal secret suffixes.
+func GetShootProjectInternalSecretSuffixes() []string {
+	return []string{
+		ShootProjectSecretSuffixCAClient,
+	}
+}
```
Comment (rfranzke) [inline pkg/utils/gardener/shoot.go:238]: Nit: Unit test?

File: pkg/utils/gardener/shoot.go [hunk 3]
```diff
@@ -251,6 +258,18 @@ func IsShootProjectSecret(secretName string) (string, bool) {
...
+// If no, it returns an empty string and <false>. Otherwise, it returns the shoot name and <true>.
+func IsShootProjectInternalSecret(secretName string) (string, bool) {
+	for _, v := range GetShootProjectInternalSecretSuffixes() {
+		if suffix := shootProjectSecretSuffix(v); strings.HasSuffix(secretName, suffix) {
+			return strings.TrimSuffix(secretName, suffix), true
+		}
+	}
+
+	return "", false
+}
```
Comment (rfranzke) [inline pkg/utils/gardener/shoot.go:271]: Nit: Unit test?

**Review by rfranzke — COMMENTED** [review]
Nice PR!

File: charts/gardener/controlplane/charts/application/templates/validatingwebhook-admission-controller.yaml (hunk 1 above)
Comment (timebertt) [reply charts/gardener/controlplane/charts/application/templates/validatingwebhook-admission-controller.yaml]: Definitely not. Good catch
